spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/client/KieClient.java

```diff
@@ -0,0 +1,192 @@
+      if (response == null) {
+        return null;
+      }
+      if (response.getStatusCode() == HttpStatus.SC_OK) {
+        revision = response.getHeader("X-Kie-Revision");
+        LOGGER.debug(response.getContent());
+        KVResponse allConfigList = JsonUtils.OBJ_MAPPER
+            .readValue(response.getContent(), KVResponse.class);
+        return getConfigByLabel(serviceCombConfigProperties, allConfigList);
+      } else if (response.getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
+        LOGGER.info(response.getStatusMessage());
+        return null;
+      } else if (response.getStatusCode() == HttpStatus.SC_NOT_MODIFIED) {
+        return null;
+      } else {
+        throw new RemoteOperationException(
+            "read response failed. status:" + response.getStatusCode() + "; message:" + response
+                .getStatusMessage() + "; content:" + response.getContent());
+      }
+    } catch (Exception e) {
+      configCenterConfig.toggle();
+      throw new RemoteOperationException("read response failed. " + response, e);
+    }
+  }
+
+
+  private Map<String, String> getConfigByLabel(
+      ServiceCombConfigProperties serviceCombConfigProperties, KVResponse resp) {
+    Map<String, String> resultMap = new HashMap<>();
+    List<KVDoc> appList = new ArrayList<>();
+    List<KVDoc> serviceList = new ArrayList<>();
+    List<KVDoc> versionList = new ArrayList<>();
+    for (KVDoc kvDoc : resp.getData()) {
+      if (!StringUtils.isEmpty(kvDoc.getStatus()) && !kvDoc.getStatus()
+          .equals(ConfigConstants.STATUS_ENABLED)) {
+        continue;
+      }
+      Map<String, String> labelsMap = kvDoc.getLabels();
+      if (labelsMap.containsKey(ConfigConstants.LABEL_APP) && labelsMap
+          .get(ConfigConstants.LABEL_APP)
+          .equals(serviceCombConfigProperties.getAppName())
+          && labelsMap.containsKey(ConfigConstants.LABEL_ENV) && labelsMap
+          .get(ConfigConstants.LABEL_ENV)
+          .equals(serviceCombConfigProperties.getEnv())) {
+        if (!labelsMap.containsKey(ConfigConstants.LABEL_SERVICE)) {
+          appList.add(kvDoc);
+        }
+        if (labelsMap.containsKey(ConfigConstants.LABEL_SERVICE) && labelsMap
+            .get(ConfigConstants.LABEL_SERVICE)
+            .equals(serviceCombConfigProperties.getServiceName())) {
+          if (!kvDoc.getLabels().containsKey(ConfigConstants.LABEL_VERSION)) {
+            serviceList.add(kvDoc);
+          }
+          if (labelsMap.containsKey(ConfigConstants.LABEL_VERSION) && labelsMap
+              .get(ConfigConstants.LABEL_VERSION)
+              .equals(serviceCombConfigProperties.getServiceName())) {
+            versionList.add(kvDoc);
+          }
+        }
+      }
+    }
+    //kv is priority
+    for (KVDoc kvDoc : appList) {
+      resultMap.putAll(processValueType(kvDoc));
+    }
+    for (KVDoc kvDoc : serviceList) {
+      resultMap.putAll(processValueType(kvDoc));
```

Comment:
This code is very long, and seams you process serviceList, versionList, etc with the same code `resultMap.putAll(processValueType(kvDoc));` , and the logic maybe better refactored.

Reply:
seems not easy to refactory